pkg/controller/master/node/maintain_controller.go

```diff
@@ -109,9 +139,24 @@ func (h *maintainNodeHandler) OnNodeChanged(_ string, node *corev1.Node) (*corev
 		}
 	}
 
-	toUpdate := node.DeepCopy()
-	toUpdate.Annotations[MaintainStatusAnnotationKey] = MaintainStatusComplete
-	return h.nodes.Update(toUpdate)
+	nodeCopy := node.DeepCopy()
+
+	nodeCopy.Annotations[MaintainStatusAnnotationKey] = MaintainStatusComplete
+	nodeCopy, err = h.nodes.Update(nodeCopy)
+	if err != nil {
+		return node, err
+	}
+
+	util.AddOrUpdateConditionToNode(nodeCopy, corev1.NodeCondition{
+		Type:               util.NodeConditionTypeMaintenanceMode,
+		Status:             corev1.ConditionTrue,
+		LastHeartbeatTime:  metav1.Now(),
+		LastTransitionTime: metav1.Now(),
+		Reason:             util.NodeConditionReasonCompleted,
+		Message:            "Draining the node is completed",
```

Comment:
```suggestion
		Message:            "Maintenance mode enabled",
```
Nit: I think technically this controller is more about enabling/disabling maintenance mode than about draining the node, right?